pkg/fission-cli/cmd/spec/list.go

```diff
@@ -0,0 +1,454 @@
+
+	kws, err := getAllKubeWatchTriggers(opts)
+	if err != nil {
+		return errors.Wrap(err, "error getting Kube Watchers from all namespaces")
+	}
+	specKubeWatchers := getSpecKubeWatchers(kws, deployID)
+	ShowAppliedKubeWatchers(specKubeWatchers)
+
+	return nil
+}
+
+func getAppliedFunctions(fns []fv1.Function, deployID string) []fv1.Function {
+	var fnlist []fv1.Function
+	if len(fns) > 0 {
+		for _, f := range fns {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				fnlist = append(fnlist, f)
+			}
+		}
+	}
+	return fnlist
+}
+
+func getAppliedEnvironments(envs []fv1.Environment, deployID string) []fv1.Environment {
+	var envlist []fv1.Environment
+	if len(envs) > 0 {
+		for _, f := range envs {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				envlist = append(envlist, f)
+			}
+		}
+	}
+	return envlist
+}
+
+func getAppliedPackages(pkgs []fv1.Package, deployID string) []fv1.Package {
+	var pkglist []fv1.Package
+	if len(pkgs) > 0 {
+		for _, f := range pkgs {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				pkglist = append(pkglist, f)
+			}
+		}
+	}
+	return pkglist
+}
+func getAppliedCanaryConfigs(canaryCfgs []fv1.CanaryConfig, deployID string) []fv1.CanaryConfig {
+	var canaryConfiglist []fv1.CanaryConfig
+	if len(canaryCfgs) > 0 {
+		for _, f := range canaryCfgs {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				canaryConfiglist = append(canaryConfiglist, f)
+			}
+		}
+	}
+	return canaryConfiglist
+}
+
+func getAppliedHTTPTriggers(hts []fv1.HTTPTrigger, deployID string) []fv1.HTTPTrigger {
+	var httpTriggerlist []fv1.HTTPTrigger
+	if len(hts) > 0 {
+		for _, f := range hts {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				httpTriggerlist = append(httpTriggerlist, f)
+			}
+		}
+	}
+	return httpTriggerlist
+
+}
+
+func getAppliedMessageQueueTriggers(mqts []fv1.MessageQueueTrigger, deployID string) []fv1.MessageQueueTrigger {
+	var mqTriggerlist []fv1.MessageQueueTrigger
+	if len(mqts) > 0 {
+		for _, f := range mqts {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				mqTriggerlist = append(mqTriggerlist, f)
+			}
+		}
+	}
+	return mqTriggerlist
+}
+func getAppliedTimeTriggers(tts []fv1.TimeTrigger, deployID string) []fv1.TimeTrigger {
+	var timeTriggerlist []fv1.TimeTrigger
+	if len(tts) > 0 {
+		for _, f := range tts {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				timeTriggerlist = append(timeTriggerlist, f)
+			}
+		}
+	}
+	return timeTriggerlist
+}
+func getSpecKubeWatchers(ws []fv1.KubernetesWatchTrigger, deployID string) []fv1.KubernetesWatchTrigger {
+	var kubeWatchTriggerlist []fv1.KubernetesWatchTrigger
+	if len(ws) > 0 {
+		for _, f := range ws {
+			if f.Metadata.Annotations["fission-uid"] == deployID {
+				kubeWatchTriggerlist = append(kubeWatchTriggerlist, f)
+			}
+		}
+	}
+	return kubeWatchTriggerlist
+}
+
+// ShowFunctions displays info of Functions
+func ShowFunctions(fns []fv1.Function) {
```

Comment:
Is it intentional to export the function? If not, how about we change it to lower case `showFunctions`?
(here and other functions.)

Reply:
Yes. It is intentional.  If we will need to print any of the resource in same format as above we can use the same function. 
I think we should export print functions of `fission fn list` or others as well. In that case we can use existing code and reduce code duplication.

Reply:
fair enough. LGTM